packages/composable-controller/src/ComposableController.ts

```diff
@@ -28,31 +31,69 @@ export type ControllerList = ControllerInstance[];
  * @param controller - Controller instance to check
  * @returns True if the controller is an instance of BaseControllerV1
  */
-function isBaseControllerV1(
+export function isBaseControllerV1(
   controller: ControllerInstance,
-  // TODO: Replace `any` with type
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-): controller is BaseControllerV1<any, any> {
-  return controller instanceof BaseControllerV1;
+): controller is BaseControllerV1<
+  BaseConfig & Record<string, unknown>,
+  BaseState & Record<string, unknown>
+> {
+  return (
+    'name' in controller &&
+    typeof controller.name === 'string' &&
+    'defaultConfig' in controller &&
+    typeof controller.defaultConfig === 'object' &&
+    'defaultState' in controller &&
+    typeof controller.defaultState === 'object' &&
+    'disabled' in controller &&
+    typeof controller.disabled === 'boolean' &&
+    controller instanceof BaseControllerV1
+  );
+}
+
+/**
+ * Determines if the given controller is an instance of BaseController
+ * @param controller - Controller instance to check
+ * @returns True if the controller is an instance of BaseController
+ */
+export function isBaseController(
+  controller: ControllerInstance,
+): controller is BaseController<never, never, never> {
+  return (
+    'name' in controller &&
+    typeof controller.name === 'string' &&
+    'state' in controller &&
+    typeof controller.state === 'object' &&
+    controller instanceof BaseController
+  );
 }
 
 export type ComposableControllerState = {
-  [name: string]: ControllerInstance['state'];
+  // `any` is used here to disable the `BaseController` type constraint which expects state properties to extend `Record<string, Json>`.
+  // `ComposableController` state needs to accommodate `BaseControllerV1` state objects that may have properties wider than `Json`.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [name: string]: Record<string, any>;
 };
 
 export type ComposableControllerStateChangeEvent = ControllerStateChangeEvent<
   typeof controllerName,
-  ComposableControllerState
+  Record<string, unknown>
 >;
 
 export type ComposableControllerEvents = ComposableControllerStateChangeEvent;
 
+type AnyControllerStateChangeEvent = ControllerStateChangeEvent<
+  string,
+  Record<string, unknown>
+>;
+
+type AllowedEvents = AnyControllerStateChangeEvent;
+
 export type ComposableControllerMessenger = RestrictedControllerMessenger<
   typeof controllerName,
   never,
-  ControllerStateChangeEvent<string, Record<string, unknown>>,
-  string,
-  string
+  ComposableControllerEvents | AllowedEvents,
+  never,
```

Comment:
No reason to keep this as `string` since ComposableController doesn't use any actions from other controllers.